src/handover.cpp

```diff
@@ -0,0 +1,222 @@
+
+        case AnyThreadConfined::Type::Results:
+            return AnyHandover(shared_group.export_for_handover(m_results.get_query(), ConstSourcePayload::Copy),
+                               m_results.get_sort());
+    }
+}
+
+AnyHandover::AnyHandover(AnyHandover&& handover)
+{
+    switch (handover.m_type) {
+        case AnyThreadConfined::Type::Object:
+            new (&m_object.row_handover) RowHandover(std::move(handover.m_object.row_handover));
+            new (&m_object.object_schema_name) std::string(std::move(handover.m_object.object_schema_name));
+            break;
+
+        case AnyThreadConfined::Type::List:
+            new (&m_list.link_view_handover) LinkViewHandover(std::move(handover.m_list.link_view_handover));
+            break;
+
+        case AnyThreadConfined::Type::Results:
+            new (&m_results.query_handover) QueryHandover(std::move(handover.m_results.query_handover));
+            new (&m_results.sort_order) SortOrder(std::move(handover.m_results.sort_order));
+            break;
+    }
+    new (&m_type) AnyThreadConfined::Type(handover.m_type);
+}
+
+AnyHandover& AnyHandover::operator=(AnyHandover&& handover)
+{
+    switch (handover.m_type) {
+        case AnyThreadConfined::Type::Object:
+            m_object.row_handover = std::move(handover.m_object.row_handover);
+            m_object.object_schema_name = std::move(handover.m_object.object_schema_name);
+            break;
+
+        case AnyThreadConfined::Type::List:
+            m_list.link_view_handover = std::move(handover.m_list.link_view_handover);
+            break;
+
+        case AnyThreadConfined::Type::Results:
+            m_results.query_handover = std::move(handover.m_results.query_handover);
+            m_results.sort_order = std::move(handover.m_results.sort_order);
+            break;
+    }
+    m_type = std::move(handover.m_type);
+    return *this;
+}
+
+AnyHandover::~AnyHandover()
+{
+    switch (m_type) {
+        case AnyThreadConfined::Type::Object:
+            m_object.row_handover.~unique_ptr();
+            break;
+
+        case AnyThreadConfined::Type::List:
+            m_list.link_view_handover.~unique_ptr();
+            break;
+
+        case AnyThreadConfined::Type::Results:
+            m_results.query_handover.~unique_ptr();
+            m_results.sort_order.~SortOrder();
+            break;
+    }
+}
+
+AnyThreadConfined AnyHandover::import_from_handover(SharedRealm realm) &&
+{
+    SharedGroup& shared_group = Realm::Internal::get_shared_group(*realm);
+    switch (m_type) {
+        case AnyThreadConfined::Type::Object: {
+            auto row = shared_group.import_from_handover(std::move(m_object.row_handover));
+            auto object_schema = realm->schema().find(m_object.object_schema_name);
+            REALM_ASSERT_DEBUG(object_schema != realm->schema().end());
+            return AnyThreadConfined(Object(std::move(realm), std::move(*object_schema), std::move(*row)));
```

Comment:
`object_schema` should not be moved here. It happens to be pointless here (`Object` takes a const ref), but if it wasn't pointless it'd be stealing the Realm's ObjectSchema.